README.md

```diff
@@ -0,0 +1,153 @@
+the underlying Kubernetes endpoint supports and the user's permissions.
+
+* `/v1/{type}` - all cluster-scoped resources OR all resources in all
+  namespaces of type `{type}` that the user has access to
+* `/v1/{type}/{name}` - cluster-scoped resource of type `{type}` and unique name `{name}`
+* `/v1/{type}/{namespace}` - all resources of type `{type}` under namespace `{namespace}`
+* `/v1/{type}/{namespace}/{name}` - resource of type `{type}` under namespace
+  `{namespace}` with name `{name}` unique within the namespace
+
+### Query parameters
+
+steve supports query parameters to perform actions or process data on top of
+what Kubernetes supports.
+
+#### `link`
+
+Trigger a link handler, which is registered with the schema. Examples are
+calling the shell for a cluster, or following logs during cluster or catalog
+operations:
+
+```
+GET /v1/management.cattle.io.clusters/local?link=log
+```
+
+#### `action`
+
+Trigger an action handler, which is registered with the schema. Examples are
+generating a kubeconfig for a cluster, or installing an app from a catalog:
+
+```
+POST /v1/catalog.cattle.io.clusterrepos/rancher-partner-charts?action=install
+```
+
+#### `filter` (new for Rancher v2.7.x)
+
+Only applicable to list requests (`/v1/{type}` and `/v1/{type}/{namespace}`)
+
+Filter results by a designated field. Filter keys use dot notation to denote
+the subfield of an object to filter on. The filter value is matched as a
+substring.
+
+Example, filtering by object name:
+
+```
+/v1/{type}?filter=metadata.name=foo
+```
+
+Filters are ANDed together, so an object must match all filters to be
+included in the list.
+
+```
+/v1/{type}?filter=metadata.name=foo&filter=metadata.namespace=bar
+```
+
+Arrays are searched for matching items. If any item in the array matches, the
+item is included in the list.
+
+```
+/v1/{type}?filter=spec.containers.image=alpine
+```
+
+#### `sort` (new for Rancher v2.7.x)
+
+Only applicable to list requests (`/v1/{type}` and `/v1/{type}/{namespace}`)
+
+Results can be sorted lexicographically by primary and secondary colums.
+
+Sorting by only a primary column, for example name:
+
+```
+/v1/{type}?sort=metadata.name
+```
+
+Reverse sorting by name:
+
+```
+/v1/{type}?sort=-metadata.name
+```
+
+Secondary sort criteria is comma separated.
+
+Example, sorting by name and creation time in ascending order:
+
+```
+/v1/{type}?sort=metadata.name,metadata.creationTimestamp
+```
+
+Reverse sort by name, normal sort by creation time:
+
+```
+/v1/{type}?sort=-metadata.name,metadata.creationTimestamp
+```
+
+Normal sort by name, reverse sort by creation time:
+
+```
+/v1/{type}?sort=metadata.name,-metadata.creationTimestamp
+```
+
+#### `page` and `pagesize` (new for Rancher v2.7.x)
+
+Only applicable to list requests (`/v1/{type}` and `/v1/{type}/{namespace}`)
+
+Results can be batched by pages for easier display.
+
+Example initial request, returns a page with 10 results:
```

Comment:
```suggestion
Example initial request returns a page with 10 results:
```

Reply:
We can use a semicolon without a conjunction if they are independent clauses.
```suggestion
Example initial request; returns a page with 10 results.
```

Reply:
Changed to "returning" to make it one clause and remove the comma